.github/workflows/third-party.yml

```diff
@@ -2,6 +2,12 @@
 # https://github.com/python/typing_extensions/blob/main/.github/workflows/third_party.yml.
 # Running the tests of various third-party libraries that use Pydantic. This helps us spot regressions early, and helps
 # flag when third-party libraries are making incorrect assumptions that might cause them to break when we cut a new release.
+# Instructions:
+# - The CI should be as close as possible to the original project's workflows.
+# - Use the checkout action to clone Pydantic with a custom path (e.g. `path: pydantic-latest`).
+# - Make sure Pydantic is installed in editable mode (e.g. `uv pip install -e ./pydantic-latest`)
+#   so that the path appears in the `pip list` output (and so we can be assured Pydantic was properly
+#   installed from the provided path).
```

Comment:
Good to add these here. There's so much redundancy across jobs, maybe we could consolidate with shared logic for installing pydantic and settings things up...